client/app/reducers/auth.reducer.js

```diff
@@ -7,18 +7,23 @@ const loginInit = {
 
 export function loginReducer(state = loginInit, action) {
   const dispatch = action.type;
+  const newState = state;
+
   if (dispatch === 'signin') {
-    state.user = action.user;
-    state.loggedIn = true;
-    return state;
+    newState.user = action.user;
+    newState.loggedIn = true;
+
+    return Object.assign({}, newState);
```

Comment:
Why not just do return Object.assign({}, state, {user: action.user, loggedIn: true}); ?